code/modules/assembly/igniter.dm

```diff
@@ -56,5 +56,12 @@
 
 
 /obj/item/assembly/igniter/attack_self(mob/user)
-	activate()
+	if(cooldown > 0)
+		return FALSE
+	cooldown = 2
+	addtimer(CALLBACK(src, PROC_REF(process_cooldown)), 10)
+	var/turf/location = get_turf(loc)
+	if(location)
+		location.hotspot_expose(1000, 1000)
+	sparks.start()
 	add_fingerprint(user)
```

Comment:
This feels like such an obscene hack for this entire ordeal.

Why are we not just checking if its part of an assembly when you attack_hand() it?

Reply:
Yeah, I suppose I was too focused on keeping the behaviour _exactly_ as it was before it got broken, when it doesn't really need to. The change does mean assemblies no longer spark when used in hand but that's not an issue.